src/Commands/core/DeployCommands.php

```diff
@@ -33,13 +34,32 @@ public function deploy(): void
         $redispatchOptions = Drush::redispatchOptions();
         $manager = $this->processManager();
 
+        // Prepare custom options for checking configuration state.
+        $configStatusOptions = $redispatchOptions;
+        $configStatusOptions['format'] = 'php';
+        // Record the current state of configuration.
+        $process = $manager->drush($self, 'config:status', [], $configStatusOptions);
+        $process->mustRun();
+        $originalConfigDiff = unserialize($process->getOutput());
+
         $this->logger()->notice("Database updates start.");
         $options = ['no-cache-clear' => true];
         $process = $manager->drush($self, 'updatedb', [], $options + $redispatchOptions);
         $process->mustRun($process->showRealtime());
 
         $this->cacheRebuild($manager, $self, $redispatchOptions);
 
+        // Record the current state of configuration.
+        $process = $manager->drush($self, 'config:status', [], $configStatusOptions);
+        $process->mustRun();
+        $newConfigDiff = unserialize($process->getOutput());
```

Comment:
Curious what the motivation is to use JSON here given that this is all PHP tooling and a php native format will process more efficiently, 

JSON makes sense if you want to serialize between languages but I don't see the reason for it here.

Reply:
> php native format will process more efficiently,

Do you have a source for that claim? json_encode()/json_decode() are C functions and should process just as fast as serialize() and unserialize(). the motivation is that we already have a getOutputAsJson() function so using JSON saves some code (i.e. the unserialize() call).

Reply:
Here's an actual test result for versions of PHP from 5.4 to 8.27: https://3v4l.org/qFKvT

I don't see how the language (C) would determine the efficiency of a routine rather than the implementation of the routine. But I digress :) 

Thank you for prompting me to re-examine my previous understanding of the matter. Because while my comment came from a pure performance consideration as well as the desire to push against the fashion for JSON at the moment, there are other things to consider as well, and one of those is safety.  While `unserialize` does perform better, there is a slight drawback with using unserialize with regards to execution safety of the encoded data. I'm not sure how relevant that is in this case.

If we were trying to unserialize objects, PHP's native unserializer would be the obvious winner as `json_decode` cannot restore PHP class types (nor magic methods and protected/private properties). But that's not what we are doing here. 

If we were exporting data that needed to be portable, JSON (or xml) would be the obvious choice rather than php serialized data. 

I guess it doesn't really matter. Raw speed would tilt in favor of `unserialize`. But if the drush community just generally prefers JSON for data portability reasons, that's a good enough argument for me.